connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -547,32 +531,16 @@ public void testReadEndOffsetsUsingAdminThatFailsWithRetriable() throws Exceptio
     private void setupWithAdmin() {
         Supplier<TopicAdmin> adminSupplier = () -> admin;
         java.util.function.Consumer<TopicAdmin> initializer = admin -> { };
-        store = PowerMock.createPartialMock(KafkaBasedLog.class, new String[]{"createConsumer", "createProducer"},
-                TOPIC, PRODUCER_PROPS, CONSUMER_PROPS, adminSupplier, consumedCallback, time, initializer);
-    }
-
-    private void expectProducerAndConsumerCreate() throws Exception {
-        PowerMock.expectPrivate(store, "createProducer")
-                 .andReturn(producer);
-        PowerMock.expectPrivate(store, "createConsumer")
-                 .andReturn(consumer);
+        store = new MockedKafkaBasedLog(TOPIC, PRODUCER_PROPS, CONSUMER_PROPS, adminSupplier, consumedCallback, time, initializer);
     }
 
-    private void expectStart() throws Exception {
-        initializer.run();
-        EasyMock.expectLastCall().times(1);
-
-        expectProducerAndConsumerCreate();
+    private void expectStart() {
+        verify(initializer).accept(any());
     }
 
     private void expectStop() {
-        producer.close();
-        PowerMock.expectLastCall();
+        verify(producer).close();
         // MockConsumer close is checked after test.
```

Comment:
I think we can just fold this logic into the method here since it's used consistently throughout each test case that invokes this method:
```
        assertFalse(store.thread.isAlive());
        assertTrue(consumer.closed());
```